fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -636,9 +642,38 @@ protected void replaceResourceWithStream(final FedoraResource resource,
             throw new RepositoryRuntimeException(e);
         }
 
+        ensureConsistencyOfInput(inputModel);
+
         resource.replaceProperties(translator(), inputModel, resourceTriples);
     }
 
+    private void ensureConsistencyOfInput(final Model inputModel) throws BadRequestException {
+        LOGGER.info("ensuring consistency!");
+        // check that ldp:hasMemberRelation value is not server managed predicate.
+        final StmtIterator iterator = inputModel.listStatements();
+        LOGGER.info("iterator.hasNext()={}", iterator.hasNext());
+
+        while (iterator.hasNext()) {
+            final Statement s = iterator.next();
+            LOGGER.debug("statement: s={}, p={}, o={}", s.getSubject(), s.getPredicate(), s.getObject());
+            if (s.getPredicate().equals(HAS_MEMBER_RELATION)) {
+                final RDFNode obj = s.getObject();
+                if (obj.isResource()) {
+                    final String uri = obj.asResource().getURI();
+                    RdfLexicon.serverManagedProperties.forEach(p -> {
```

Comment:
I suspect there is a snappy one-liner using the Predicate, but can't this at least be shortened to: `if (RdfLexicon.serverManagedProperties.contains(p)) ...`